ferrocene/doc/sphinx-shared-resources/exts/ferrocene_domain_cli/domain.py

```diff
@@ -18,17 +22,28 @@ class ProgramDirective(SphinxDirective):
     has_content = True
     required_arguments = 1
     final_argument_whitespace = True
+    option_spec = {"no_traceability_matrix": directives.flag}
 
     def run(self):
+        # if there already is program data in storage, a ProgramDirective is
+        # within a ProgramDirective, which isn't supported
         if PROGRAM_STORAGE in self.env.temp_data:
             warn("cli:program inside cli:program isn't supported", self.get_location())
             return []
-        self.env.temp_data[PROGRAM_STORAGE] = self.arguments[0]
 
+        # store arguments, so they can be accessed by child `OptionDirective`s
+        self.env.temp_data[PROGRAM_STORAGE] = (
+            self.arguments[0],
+            "no_traceability_matrix" in self.options,
+        )
```

Comment:
Nit: can we define a dataclass here, so that we can have named fields for program name and no_traceability_matrix?

Reply:
Sure